src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -249,23 +248,28 @@ class JesBackend extends Backend with LazyLogging with ProductionJesAuthenticati
   def generateJesOutputs(backendCall: BackendCall): Seq[JesOutput] = {
     val wdlFileOutputs = backendCall.call.task.outputs flatMap { taskOutput =>
       taskOutput.expression.evaluateFiles(backendCall.lookupFunction, new NoFunctions, taskOutput.wdlType) match {
-        case Success(wdlFiles) => wdlFiles map gcsPathToLocal map (_.valueString)
+        case Success(wdlFiles) => wdlFiles map gcsPathToLocal
         case Failure(ex) =>
           logger.warn(s"${makeTag(backendCall)} Could not evaluate $taskOutput: ${ex.getMessage}")
           Seq.empty[String]
       }
     }
-    wdlFileOutputs.distinct map { filePath =>
-      JesOutput(makeSafeJesReferenceName(filePath), s"${backendCall.callGcsPath}/$filePath", localFilePathFromRelativePath(filePath))
+
+    // Create the mappings. GLOB mappings require special treatment (i.e. stick everything matching the glob in a folder)
+    wdlFileOutputs.distinct map {
+      file => file match {
+        case WdlFile(filePath, GlobOrNot.NotGlob) => JesOutput(makeSafeJesReferenceName(filePath), s"${backendCall.callGcsPath}/$filePath", localFilePathFromRelativePath(filePath))
+        case WdlFile(filePath, GlobOrNot.YesGlob) => JesOutput(makeSafeJesReferenceName(filePath), s"${backendCall.callGcsPath}/globbed-${DigestionUtil.md5Sum(filePath)}/", localFilePathFromRelativePath(filePath))
```

Comment:
Most of what goes into the `JesOutput` is the same in both branches.  Perhaps only do the different stuff in the cases and then assemble into the `JesOutput`?